src/extendedBitStream.ts

```diff
@@ -0,0 +1,86 @@
+import { BitStream } from "bit-buffer";
+
+export default class ExtendedBitStream extends BitStream {
+
+    constructor(buffer: Buffer) {
+        super(buffer, buffer.byteOffset, buffer.byteLength);
+    }
+
+    public swapEndian() {
+        this.bigEndian = !this.bigEndian;
+    }
+
+    // the type definition for BitStream does not include the _index property
+    // since it's supposed to be private, but it's needed 4 times here sooo
```

Comment:
The type definition can be augmented to include this, or you can probably try to just use `index`?